### PR TITLE
New Structure for the Plugin Help Pages

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#39832](https://central.tri.be/issues/39832)
_Rel:_ moderntribe/the-events-calendar#407

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Help_Page.php:302](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L302)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Help_Page.php:393](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L393)
:x: | Echoed strings should not be bracketed | [Help_Page.php:422](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L422)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$link' | [Help_Page.php:422](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L422)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$link' | [Help_Page.php:422](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L422)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$addon' | [Help_Page.php:436](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Admin/Help_Page.php#L436)
:x: | String &quot;Sometimes when troubleshooting a problem it is helpful to review the details of your system&rsquo;s set-up.&quot; does not require double quotes; use single quotes instead | [Support.php:46](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Support.php#L46)
:x: | String &quot;For your convenience we&rsquo;ve put together this report. If you are posting in our premium forums, please copy and paste this information into the System Information field. That will help us help you faster!&quot; does not require double quotes; use single quotes instead | [Support.php:47](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/Tribe/Support.php#L47)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [tribe-options-help.php:60](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L60)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$help' | [tribe-options-help.php:129](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L129)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [tribe-options-help.php:129](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L129)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$item' | [tribe-options-help.php:129](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L129)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$help' | [tribe-options-help.php:131](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L131)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$more_text' | [tribe-options-help.php:131](https://github.com/moderntribe/tribe-common/blob/17d58ead4b50ed6271fd2b87ff558c5d78bdfd0c/src/admin-views/tribe-options-help.php#L131)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Admin/Help_Page.php [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+			$plugins['the-events-calendar']['is_active'] = true;
+		}
+
+		$plugins['event-tickets'] = array(
+			'name' => 'event-tickets',
+			'title' => __( 'Event Tickets', 'tribe-common' ),
+			'api' => 'http://wpapi.org/api/plugin/event-tickets.php',
+			'repo' => 'http://wordpress.org/extend/plugins/event-tickets/',
+			'stars_url' => 'http://wordpress.org/support/view/plugin-reviews/event-tickets?filter=5',
+			'description' => __( 'Events Tickets is a carefully crafted, extensible plugin that lets you easily sell tickets for your events.', 'tribe-common' ),
```
Comment (zbtirrell) [inline src/Tribe/Admin/Help_Page.php]: this text will need an adjustment.

File: src/Tribe/Admin/Help_Page.php [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+			$plugins['event-tickets']['is_active'] = true;
+		}
+
+		$plugins['advanced-post-manager'] = array(
+			'name' => 'advanced-post-manager',
+			'title' => __( 'Advanced Post Manager', 'tribe-common' ),
+			'api' => 'http://wpapi.org/api/plugin/advanced-post-manager.php',
+			'repo' => 'http://wordpress.org/extend/plugins/advanced-post-manager/',
+			'stars_url' => 'http://wordpress.org/support/view/plugin-reviews/advanced-post-manager?filter=5',
+			'description' => null,
```
Comment (zbtirrell) [inline src/Tribe/Admin/Help_Page.php]: "Turbo charge your posts admin for any custom post type with sortable filters and columns, and auto-registration of metaboxes."

File: src/Tribe/Admin/Help_Page.php [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+
+		if ( class_exists( 'Tribe_APM' ) ){
+			$plugins['advanced-post-manager']['is_active'] = true;
+		}
+
+		$plugins = (array) apply_filters( 'tribe_help_plugins', $plugins );
+
+		// Only active ones?
+		if ( true === $is_active ){
+			foreach ( $plugins as $key => $plugin ) {
```
Comment (zbtirrell) [inline src/Tribe/Admin/Help_Page.php]: `$plugin` here is potentially confusing as there is already a `$plugin` variable in scope

File: src/Tribe/Admin/Help_Page.php [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+
+		$addons[] = array(
+			'title' => esc_html__( 'Events Filter Bar', 'tribe-common' ),
+			'link'  => 'http://m.tri.be/hu',
+			'plugin' => array( 'the-events-calendar' ),
+			'is_active' => class_exists( 'Tribe__Events__Filterbar__View' ),
+		);
+
+		$addons[] = array(
+			'title' => esc_html__( 'Events Tickets Plus', 'tribe-common' ),
```
Comment (zbtirrell) [inline src/Tribe/Admin/Help_Page.php]: "Event Tickets Plus"

File: src/Tribe/Admin/Help_Page.php [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+			if ( 'install' !== $status['status'] && ! $plugin_active ){
+				$args = array(
+					'action' => 'activate',
+					'plugin' => $status['file'],
+					'plugin_status' => 'all',
+					'paged' => 1,
+					's' => '',
+				);
+				$activate_url = wp_nonce_url( add_query_arg( $args, 'plugins.php' ), 'activate-plugin_' . $status['file'] );
+				$link = '<a class="button" href="' . $activate_url . '" aria-label="' . esc_attr( sprintf( __( 'Activate %s', 'tribe-common' ), $plugin->name ) ) . '">' . esc_html__( 'Activate Plugin', 'tribe-common' ) . '</a>';
```
Comment (zbtirrell) [inline src/Tribe/Admin/Help_Page.php]: `aria-label` ?

Comment (bordoni) [inline src/Tribe/Admin/Help_Page.php]: This was on WordPress core Code, just copied. I think it's something for screen readers.

File: src/Tribe/Admin/Help_Page.php (hunk 3 above)
Comment (bordoni) [inline src/Tribe/Admin/Help_Page.php]: Changed the argument, because it makes more sense here.

File: src/Tribe/Admin/Help_Page.php [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+	 * @param  string  $plugin_name    Should get only one plugin?
+	 * @param  boolean $is_active Only get active plugins?
+	 * @return array
+	 */
+	public function get_plugins( $plugin_name = null, $is_active = true ) {
+		$plugins = array();
+
+		$plugins['the-events-calendar'] = array(
+			'name' => 'the-events-calendar',
+			'title' => __( 'The Events Calendar', 'tribe-common' ),
```
Comment (borkweb) [inline src/Tribe/Admin/Help_Page.php]: `esc_html__()` for this and other lines.

File: src/Tribe/Admin/Help_Page.php [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+			'api' => 'http://wpapi.org/api/plugin/the-events-calendar.php',
+			'repo' => 'http://wordpress.org/extend/plugins/the-events-calendar/',
+			'stars_url' => 'http://wordpress.org/support/view/plugin-reviews/the-events-calendar?filter=5',
+			'description' => __( 'The Events Calendar is a carefully crafted, extensible plugin that lets you easily share your events.', 'tribe-common' ),
+			'is_active' => false,
+			'version' => null,
+		);
+
+		if ( class_exists( 'Tribe__Events__Main' ) ){
```
Comment (borkweb) [inline src/Tribe/Admin/Help_Page.php]: Minor nitpick (rinse and repeat for other if statements): `) {` vs `){`

We really need to get a rule in place to sniff this.

File: src/Tribe/Admin/Help_Page.php [hunk 8]
```diff
@@ -0,0 +1,451 @@
...
+
+			<?php
+			if ( ! empty( $plugin->description ) && ! $plugin->is_active ) {
+				echo wpautop( $plugin->description );
+			}
+			?>
+
+			<?php if ( $api_data ) { ?>
+			<div>
+				<b><?php esc_html_e( 'Latest Version:', 'tribe-common' ); ?></b> <?php echo esc_html( $api_data->version ); ?>
```
Comment (borkweb) [inline src/Tribe/Admin/Help_Page.php]: I feel like this `<div>`/`<b>`/`<br>` listing of meta information would be better presented using `<dl>`/`<dt>`/`<dd>` elements.

Comment (bordoni) [inline src/Tribe/Admin/Help_Page.php]: It really would be, I will test that. :wink:

File: src/Tribe/Admin/Help_Page.php (hunk 7 above)
Comment (bordoni) [inline src/Tribe/Admin/Help_Page.php]: Indeed.

**Comment by bordoni** [comment]
Just modified the PR @borkweb

**Comment by borkweb** [comment]
:+1:

**Pull request merged**